### PR TITLE
Dont send notification if notification type is off in setting

### PR DESCRIPTION
**Pull request by mohammadranjbarz**
related to https://github.com/Giveth/giveth-dapps-v2/issues/709

* Consider notification setting and just create notification if it's not turned off for user
* Add icon to notification types, frontend guys need that
* Call segment is `sendSegment` is true in notification body
* Adding segment validators

The tests passed but we should deploy this PR and https://github.com/Giveth/impact-graph/pull/713 together to see wether integration works well or not

### CODE REVIEW

**Review by CarlosQ96 — CHANGES_REQUESTED** [review]
Thanks Mohammad I think just add the error and we can merge.

File: src/controllers/v1/notificationsController.ts [hunk 1]
```diff
@@ -78,7 +78,18 @@ export class NotificationsController {
...
         throw new Error(errorMessages.INVALID_NOTIFICATION_TYPE);
       }
-      let emailStatus = body.sendEmail
+      const notificationSetting =
+        await findNotificationSettingByNotificationTypeAndUserAddress({
+          notificationTypeId: notificationType.id,
+          userAddressId: userAddress.id,
+        });
+      if (!notificationSetting) {
+        //TODO I dont know what should we do in this case
```
[the remark this answers is not in the document]
Comment (CarlosQ96) [reply src/controllers/v1/notificationsController.ts:87]: We can log the error to sentry or our logs and not interrupt the process better.

[the remark this answers is not in the document]
Comment (mohammadranjbarz) [reply src/controllers/v1/notificationsController.ts:87]: I dont know what happens if we add a notification type, and user is already in DB, I think about edge cases, so I don't throw exception just add some log to can see if it happend

File: src/controllers/v1/notificationsController.ts [hunk 2]
```diff
@@ -91,19 +102,33 @@ export class NotificationsController {
...
-          notificationType.emailNotificationId!,
-          body.analyticsUserId,
-          body.segmentData,
-          body.anonymousId,
-        );
+        const segmentData = Object.assign({}, body.segment?.payload);
+        validateWithJoiSchema(segmentData, segmentValidator);
+        if (!shouldSendEmail) {
+          // Segment and autopilot will send email automatically so now we just can cancel sending event by make email empty
+          delete segmentData['email'];
```
[the remark this answers is not in the document]
Comment (mohammadranjbarz) [reply src/controllers/v1/notificationsController.ts:109]: Yeah, let hope it works :))
We should test it to see whether works or not

**Pull request merged**